src/optparse.coffee

```diff
@@ -25,37 +25,26 @@ exports.OptionParser = class OptionParser
   # parsers that allow you to attach callback actions for every flag. Instead,
   # you're responsible for interpreting the options object.
   parse: (args) ->
-    options = arguments: []
-    skippingArgument = no
-    originalArgs = args
-    args = normalizeArguments args
-    for arg, i in args
-      if skippingArgument
-        skippingArgument = no
-        continue
-      if arg is '--'
-        pos = originalArgs.indexOf '--'
-        options.arguments = options.arguments.concat originalArgs[(pos + 1)..]
+    state =
+      argsLeft: args[..]
+      options: {}
+    while (arg = state.argsLeft.shift())?
+      if (arg.match(LONG_FLAG) ? arg.match(SHORT_FLAG))?
+        tryMatchOptionalArgument(arg, state, @rules)
+      else if (multiMatch = arg.match(MULTI_FLAG))?
+        # Normalize arguments by expanding merged flags into multiple
+        # flags. This allows you to have `-wl` be the same as `--watch --lint`.
+        normalized = "-#{multiArg}" for multiArg in multiMatch[1].split ''
+        state.argsLeft.unshift(normalized...)
+      else
+        # the CS option parser is a little odd; options after the first
```

Comment:
Got it, will fix. The first two lines are actually from the original source, so I wasn't sure whether I should change them.